packages/@aws-cdk/aws-lambda/lib/runtime.ts

```diff
@@ -9,18 +9,18 @@ export interface LambdaRuntimeProps {
 /**
  * Lambda function runtime environment.
  */
-export class LambdaRuntime {
+export class LambdaRuntime implements InlinableLambdaRuntime, InlinableJavascriptLambdaRuntime {
```

Comment:
Shouldn't we just get rid of these interfaces? Nothing takes them as arguments, `Lambda` just takes a `LambdaRuntime` argument, and since `LambdaRuntime` already exposes the `supportsInlineCode` property that code is looking for, we're already done.

Reply:
It is currently used by the (soon to be removed) [`InlineJavaScriptLambda`](https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/inline.ts#L71)

Reply:
I want to do a bit of overhaul on Lambda soon, but in the meantime, I just want to unblock Java users from using Lambda.

Reply:
In that case I would feel better if you turned `InlineJavaScriptLambda` into a class and used the pattern I described in the other comments. That restores the type safety and still compiles properly:

```ts
    public static get NodeJS(): InlinableJavascriptLambdaRuntime {
        return new InlinableJavascriptLambdaRuntime('nodejs');
    }
```

I will approve in any case; do what you deem best.